SU2_CFD/include/solvers/CSolver.hpp

```diff
@@ -4411,6 +4411,24 @@ class CSolver {
    */
   inline virtual bool GetHasHybridParallel() const { return false; }
 
+  /*!
+   * \brief Get the massflow of the streamwise periodic donor/outlet boundary.
+   * \return The streamwise periodic donor/outlet massflow.
+   */
+  virtual su2double GetStreamwise_Periodic_MassFlow() const { return 0.0; }
+
+  /*!
+   * \brief Get the net sum of the heatflow into the domain.
+   * \return The net sum of the heatflow into the domain.
+   */
+  virtual su2double GetStreamwise_Periodic_IntegratedHeatFlow() const { return 0.0; }
+
+  /*!
+   * \brief Get the value of the area avg periodic inlet Temperature.
+   * \return Temperature value.
+   */
+  virtual su2double GetStreamwise_Periodic_InletTemperature(void) const { return 0.0; }
```

Comment:
For gathering those 3+1(from the config which I'll add): do you have a preference of struct, tuple, or sth else? and why a solution might be preferred (if you have the time to give a rough idea)

Reply:
A struct defined in option_structure close to the streamwise enum, so that you can access things by name.
With tuples or arrays you would need to know the position that corresponds with what, and that is less readable IMO.

Reply:
But in the past I have also used "out params" to "return" multiple things from these methods...
Now I think using small custom types gives a more modular feel to things.

Reply:
is there a way to return s.th in a virtual function which is  like nullptr for pointer return types or 0.0 for su2doubles. 
```c++
virtual mystructtype() const { return  ?? }
```
I briefly searched but didn't find a good way. And having that return already a a member of CSolver is not exactly what I want

Reply:
Not in C++11 I don't think (https://en.cppreference.com/w/cpp/utility/optional).
Would returning a default constructed struct not work? The definition needs to be available to CSolver anyway so that you can access it in the output.

Reply:
I guess one can
```c++
virtual mystructtype Getmystructtype () const {mystructtype foo; return  foo; }
```
I dont get any warnings on my machine with warnlevel=3 but I am not sure whether this is illegal... and I dont want to be woken up by the Cpp-Police banging on my door tomorrow early in the morning... 'Cause thats what they tend to do
On the other hand return zeros can create nasty silent bugs as well so I will take my chances here 😎

Reply:
Ok typed that while you already answered, so i didn't see your answer right away. Is that what you refer to as `default constructed struct` ?

Reply:
return mystructtype();